framework/db/ActiveRecord.php

```diff
@@ -291,12 +303,18 @@ public static function tableName()
      */
     public static function getTableSchema()
     {
-        $schema = static::getDb()->getSchema()->getTableSchema(static::tableName());
-        if ($schema !== null) {
-            return $schema;
-        } else {
-            throw new InvalidConfigException("The table does not exist: " . static::tableName());
+        static $tableSchema = null;
+
+        if ($tableSchema === null) {
+            $schema = static::getDb()->getSchema()->getTableSchema(static::tableName());
```

Comment:
> If someone wanted to explicitly refresh the schema data, they could still call Schema::getTableSchema($refresh=true).

Yes, but your `static $tableSchema` will still contain the old instance of the `TableSchema` so `Schema::getTableSchema('mytable', true)` will have NO effect on the Active Record.